### ISSUE TITLE
No result with set_current_mission_item_async

### ISSUE THREAD

**Issue by julianoes** [mavlink/MAVSDK#1003]
There was a report that `set_current_mission_item_async` does not call the callback at all.

This needs fixing and we also need an integration tests checking it.

```
    int current = 4;
    auto prom = std::make_shared<std::promise<Mission::Result>>();
    auto future_result = prom->get_future();
    std::cout << "Setting current mission item..." << std::endl;
    mission->set_current_mission_item_async(current,
    [prom](Mission::Result result) {
        prom->set_value(result);
    });    const Mission::Result result = future_result.get(); //Wait on result    if(result == Mission::Result::SUCCESS){
        std::cout << "OK" << std::endl;
    }else{
        std::cout << "NOK" << std::endl;
    }
```

**Comment by rohan2734** [comment]
This bug doesnt seem it is for a beginner

**Comment by coderkalyan** [comment]
https://discuss.px4.io/t/mavsdk-clear-mission-async-callback-never-gets-called/17686 seems to be a report that `clear_mission_async` doesn't call the callback either. Maybe related?

**Comment by JonasVautherin** [comment]
> Maybe related?

Yes, maybe. Would you mind linking this issue from the discuss thread? :blush:

**Comment by coderkalyan** [comment]
Sure thing, will do.

**Comment by julianoes** [comment]
Fixed by #1006, and covered by an integration test in 1264.
